src/EntryPoints/MediaWikiHooks.php

```diff
@@ -32,20 +35,43 @@ static function ( SlotRoleRegistry $registry ) {
 	}
 
 	// TODO: is this the best hook?
-	public static function onOutputPageParserOutput( OutputPage $out, ParserOutput $parserOutput ): void {
-		if ( $out->getTitle()?->inNamespaces( WB_NS_ITEM, WB_NS_PROPERTY ) ) {
-			// TODO: load styles earlier because this causes the initial content to be unstyled
-			$out->addModules( 'ext.wikibase.rdf' );
+	public static function onOutputPageParserOutput( OutputPage $page, ParserOutput $parserOutput ): void {
+		$entityId = self::getEntityIdForOutputPage( $page );
 
-			$presenter = WikibaseRdfExtension::getInstance()->newStubMappingsPresenter();
-			$useCase = WikibaseRdfExtension::getInstance()->newShowMappingsUseCase( $presenter, $out->getUser() );
-			// TODO: get actual ID
-			$useCase->showMappings( new ItemId( 'Q1' ) );
+		if ( $entityId !== null ) {
+			self::addMappingUi( $page, $entityId );
+		}
+	}
+
+	private static function getEntityIdForOutputPage( OutputPage $page ): ?EntityId {
+		$title = $page->getTitle();
 
-			$out->addHTML( $presenter->getHtml() );
+		if ( $title === null
+			|| !$title->inNamespaces( WB_NS_ITEM, WB_NS_PROPERTY )
+			|| !$title->exists() ) {
+			return null;
+		}
+
+		try {
+			return WikibaseRepo::getEntityIdParser()->parse( $title->getText() );
```

Comment:
So this the right way to get the parser. Then we probably need to get rid of this:
https://github.com/ProfessionalWiki/WikibaseRDF/blob/4daf69791e6b5bfb0a37727f4f0059632d81c9c9/src/WikibaseRdfExtension.php#L77-L79

Reply:
Sure, though in our case using `BasicEntityIdParser` should not cause any problems. The main implication is that this code won't work with extra entry types added to Wikibase, but then we are explicitly filtering on items and properties already.